Examples/Jalilpour/DeliveryFactory.cs

```diff
@@ -0,0 +1,92 @@
+﻿
+namespace Examples.Jalilpour
+{
+    enum CargoType
+    {
+        Air,
+        Ship,
+        Train
+    }
+
+    public class DeliveryService
+    {
+        public Delivery Create(CargoType cargoType)
+        {
+            if (cargoType == CargoType.Air)
```

Comment:
you have created the factory for each type, but why are you making instances by your self?
you have to seperate concern to each factory to make instances. and manage the instances. 
for example : if Air must be single instance the factory of Air is responsible to do it.

Reply:
Off Course, I'm going to correct that.